### PR TITLE
added NetworkStatus package & injected it into network instrumentation

### PR DESCRIPTION
**Pull request by bryce-b**
implements attributes defined here : https://github.com/open-telemetry/opentelemetry-specification/pull/1759

I think I need to re-generate the protobuf spec as well?

### CODE REVIEW

**Comment by bryce-b** [comment]
Seems like there is a permissions issue using another github project. Will it be an issue to use a 3rd party framework here?

**Comment by nachoBonafonte** [comment]
I did an initial review it looks generally good, some minor things only mostly stylistic.

**Comment by nachoBonafonte** [comment]
> Seems like there is a permissions issue using another github project. Will it be an issue to use a 3rd party framework here?

What do you mean by permissions issue, license related? While the third party project has a compatible license there should be no problem.

**Comment by bryce-b** [comment]
>  What do you mean by permissions issue, license related? While the third party project has a compatible license there should be no problem.

The build is failing due to unable to access the `reachability` repo

**Comment by nachoBonafonte** [comment]
> >  What do you mean by permissions issue, license related? While the third party project has a compatible license there should be no problem.
> 
> 
> 
> The build is failing due to unable to access the `reachability` repo
> 
> 

I think the problem is in the URL used in Package.swift for the repo, can you try using: `https://github.com/ashleymills/Reachability.swift` ?

File: Tests/InstrumentationTests/NetworkStatusTests/NetworkStatusTests.swift [hunk 1]
```diff
@@ -0,0 +1,152 @@
+// Copyright © 2021 Elasticsearch BV
```
Comment (SergeyKanzhelev) [inline Tests/InstrumentationTests/NetworkStatusTests/NetworkStatusTests.swift:1]: is the different copyright intentional?

Comment (bryce-b) [reply Tests/InstrumentationTests/NetworkStatusTests/NetworkStatusTests.swift:1]: Oops! usually I catch this. I'll correct it right away.

**Review by nachoBonafonte — APPROVED** [review]

**Pull request merged**